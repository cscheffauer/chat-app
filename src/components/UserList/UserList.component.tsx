import React from 'react';
import UserListEntry from '../UserListEntry/UserListEntry.component';

import './UserList.scss';

interface Props {
	userList: Object;
}

const UserList = ({ userList }: Props) => {
	return (
		<div className='userlist'>
			{Object.entries(userList).map(([key, user]) => (
				<UserListEntry key={key} username={user.username} />
			))}
		</div>
	);
};

export default UserList;
