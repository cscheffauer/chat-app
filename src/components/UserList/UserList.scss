@import '../../utils/variables';

.userlist {
	height: 84vh; /* Fallback for browsers that do not support Custom Properties */
	height: calc(var(--vh, 1vh) * 84);
	background-color: $color-chat-window;
	border-left: $border-style-tabs;
	border-right: $border-style-tabs;
}
