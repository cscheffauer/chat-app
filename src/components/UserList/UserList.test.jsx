import React from 'react';
import { mount } from 'enzyme';
import UserList from './UserList.component';

describe('<UserList />', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = mount(<UserList userList={{ 1: { username: 'Max Mustermann' }, 2: { username: 'John Doe' } }} />);
	});

	afterEach(() => {
		jest.clearAllMocks();
	});
	it('expect to render UserList component', () => {
		expect(wrapper).toMatchSnapshot();
	});

	describe('Testing the UserList', () => {
		it('expect to have the usernames set in the userlist entry', () => {
			const firstUserListEntry = wrapper.find('UserListEntry').at(0);
			expect(firstUserListEntry.props().username).toBe('Max Mustermann');
			expect(firstUserListEntry.text()).toBe('Max Mustermann');

			const secondUserListEntry = wrapper.find('UserListEntry').at(1);
			expect(secondUserListEntry.props().username).toBe('John Doe');
			expect(secondUserListEntry.text()).toBe('John Doe');
		});
	});
});
