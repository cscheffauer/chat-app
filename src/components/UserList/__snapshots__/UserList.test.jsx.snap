// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserList /> expect to render UserList component 1`] = `
<UserList
  userList={
    Object {
      "1": Object {
        "username": "Max Mustermann",
      },
      "2": Object {
        "username": "John Doe",
      },
    }
  }
>
  <div
    className="userlist"
  >
    <UserListEntry
      key="1"
      username="Max Mustermann"
    >
      <div
        className="userlistentry"
      >
        <p
          className="username"
        >
          Max Mustermann
        </p>
      </div>
    </UserListEntry>
    <UserListEntry
      key="2"
      username="John Doe"
    >
      <div
        className="userlistentry"
      >
        <p
          className="username"
        >
          John Doe
        </p>
      </div>
    </UserListEntry>
  </div>
</UserList>
`;
